src/Symfony/Component/HttpClient/CachingHttpClient.php

```diff
@@ -11,62 +11,346 @@
+
+                    $maxAge = $this->determineMaxAge($headers, $cacheControl);
+
+                    $this->cache->saveDeferred(
+                        $this->cache->getItem($metadataKey)
+                            ->tag($fullUrlTag)
+                            ->set([
+                                'status_code' => $context->getStatusCode(),
+                                'headers' => $headers,
+                                'initial_age' => (int) ($headers['age'][0] ?? 0),
+                                'stored_at' => time(),
+                                'expires_at' => $this->calculateExpiresAt($maxAge),
+                                'chunks_count' => $chunkIndex,
+                            ])
+                            ->expiresAt($expiresAt)
+                    );
+
+                    $this->cache->commit();
+
+                    yield $chunk;
+
+                    return;
+                }
+
+                ++$chunkIndex;
+                $chunkKey = "{$metadataKey}_chunk_{$chunkIndex}";
+                $chunkItem = $this->cache->getItem($chunkKey)
+                    ->tag($fullUrlTag)
+                    ->set($chunk->getContent())
+                    ->expiresAt($expiresAt);
+
+                $this->cache->save($chunkItem);
+
+                yield $chunk;
+            }
+        );
+    }
+
+    public function stream(ResponseInterface|iterable $responses, ?float $timeout = null): ResponseStreamInterface
+    {
+        if ($responses instanceof ResponseInterface) {
+            $responses = [$responses];
+        }
+
+        $mockResponses = [];
+        $asyncResponses = [];
+        $clientResponses = [];
+
+        foreach ($responses as $response) {
+            if ($response instanceof MockResponse) {
+                $mockResponses[] = $response;
+            } elseif ($response instanceof AsyncResponse) {
+                $asyncResponses[] = $response;
+            } else {
+                $clientResponses[] = $response;
+            }
+        }
+
+        if (!$mockResponses && !$clientResponses) {
+            return $this->asyncStream($asyncResponses, $timeout);
+        }
+
+        if (!$asyncResponses && !$clientResponses) {
+            return new ResponseStream(MockResponse::stream($mockResponses, $timeout));
+        }
+
+        if (!$asyncResponses && !$mockResponses) {
+            return $this->client->stream($clientResponses, $timeout);
+        }
+
+        return new ResponseStream((function () use ($mockResponses, $asyncResponses, $clientResponses, $timeout) {
+            yield from MockResponse::stream($mockResponses, $timeout);
+            yield $this->client->stream($clientResponses, $timeout);
+            yield $this->asyncStream($asyncResponses, $timeout);
+        })());
+    }
+
+    public function legacyRequest(string $method, string $url, array $options = []): ResponseInterface
```

Comment:
should be private. there is no reason to introduce it as a public method (especially one that is not usable whne the object is constructed in a non-deprecated way)

Reply:
adressed in https://github.com/symfony/symfony/pull/59576/commits/806ddf90397d90569d5fe9e4d6fd5477969cbae7